libs/ui-lib/lib/cim/services/CustomManifestService.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import {
+  K8sResourceCommon,
+  Patch,
+  useK8sWatchResources,
+} from '@openshift-console/dynamic-plugin-sdk';
+import { AgentClusterInstallK8sResource } from '../types';
+import { load } from 'js-yaml';
+import { patchResource, reconcileResources } from './apis/utils';
+import { AgentClusterInstallModel, MachineConfigModel } from './apis/models';
+
+export const CustomManifestService = {
+  useCustomManifests: (agentClusterInstall?: AgentClusterInstallK8sResource) => {
+    const params = Object.fromEntries(
+      [...(agentClusterInstall?.spec?.manifestsConfigMapRefs || [])]?.map((manifest) => [
+        manifest.name,
+        {
+          name: manifest.name,
+          groupVersionKind: {
+            group: 'machineconfiguration.openshift.io',
+            version: 'v1',
+            kind: 'MachineConfig',
+          },
+        },
+      ]),
+    );
+
+    const manifestsData = useK8sWatchResources(params);
+    const customManifests = React.useMemo(
+      () => Object.values(manifestsData).map((manifest) => manifest.data),
+      [manifestsData],
+    );
+    const isLoading = Object.values(manifestsData).some((manifest) => manifest.loaded === false);
+
+    return { customManifests, isLoading };
+  },
+  onSyncCustomManifests: async (
+    agentClusterInstall?: AgentClusterInstallK8sResource,
+    values?: {
+      manifests: {
+        manifestYaml: string;
+        filename?: string;
+        created: boolean;
+      }[];
+    },
+    existingManifests?: K8sResourceCommon[],
+    onFulfill?: () => void | Promise<void>,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    onReject?: (e: any) => void,
```

Comment:
Lets use `unknown` instead of `any`
```suggestion
    onReject?: (e: unknown) => void,
```